inventory/src/main/java/org/flyve/inventory/categories/Category.java

```diff
@@ -119,10 +120,17 @@ public String getTagName() {
     public CategoryValue put(String key, CategoryValue value) {
        //Do not add value if it's null, blank or "unknow"
        if (value != null) {
-           if (value.getCategory() != null || !value.getValue().equals("") && !value.getValue().equals(Build.UNKNOWN)) {
-               return super.put(key, value);
+           if (value.getCategory() == null) {
+               String s = value.getValue();
+               if (s != null && !s.equals("") && !s.equals(Build.UNKNOWN)){
+                   return super.put(key, value);
+               } else if (value.getValues() != null && value.getValues().size() > 0) {
+                   return super.put(key, value);
+               } else {
+                   return null;
+               }
            } else {
-               return null;
+               return super.put(key, value);
            }
        } else {
     	   return null;
```

Comment:
possible NullPointerException

Reply:
That method always returns null if the condition no is true